nsv13/code/modules/munitions/ship_weapons/energy_weapons/subspacecooling.dm

```diff
@@ -0,0 +1,107 @@
+/obj/machinery/cooling
+	name = "subspace unit"
+	desc = "A subspace unit."
+	icon = 'nsv13/icons/obj/subspace.dmi'
+	icon_state = "cooler"
+	circuit = /obj/item/circuitboard/machine
+	bound_width = 32
+	pixel_x = 0
+	pixel_y = 0
+	idle_power_usage =  2000
+	var/obj/machinery/ship_weapon/energy/parent
+	var/on = FALSE
+	density = TRUE
+	critical_machine = TRUE
+
+/obj/machinery/cooling/cooler
+	name = "subspace cooling unit"
+	desc = "A cooling unit that dumps the massive amounts of heat energy weapons generate into subspace."
+	circuit = /obj/item/circuitboard/machine/cooling
+
+
+
+/obj/machinery/cooling/cooler/Initialize(mapload)
+	. = ..()
+	var/obj/machinery/ship_weapon/energy/E = locate(/obj/machinery/ship_weapon/energy) in orange(1, src) //I have no idea what I'm doing and this causes errors so
+	E.coolers |= src
+	parent = E
+
+/obj/machinery/cooling/storage/Initialize(mapload)
+	. = ..()
+	var/obj/machinery/ship_weapon/energy/E = locate(/obj/machinery/ship_weapon/energy) in orange(1, src)
+	E.storages |= src
+	parent = E
+
+
+/obj/machinery/cooling/cooler/Destroy()
+  parent.coolers -= src
+  . = ..()
+
+/obj/machinery/cooling/storage/Destroy()
+  parent.storages -= src
+  . = ..()
+
+/obj/machinery/cooling/attack_hand(mob/user)
+	. = ..()
+	if(panel_open)
+		to_chat(user, "<span class='notice'>You must turn close the panel on [src] before turning it on.</span>")
+		return
+	to_chat(user, "<span class='notice'>You press [src]'s power button.</span>")
+	on = !on
+	update_icon()
+
+/obj/machinery/cooling/examine()
+	. = ..()
+	if(on)
+		. += "The power is on"
+	if(parent)
+		. += "The thermal subspace transcever is linked"
+	else
+		. += "it's completely inactive"
```

Comment:
Can be on and also inactive which will print both messages that are somewhat contradictory. Missed an `else if` above?

Reply:
These are mostly debug messages, so printing both is still helpful even if confusing

Reply:
despite this, should I add additional handling or re-phrase it?

Reply:
Given it is currently present as the PR is testmerged, it would be useful if examine information was precise.

Reply:
I'll see if I can add additional handling

Reply:
this is basically all the existing states FOR the cooling machines, so I'm not sure that they can actually have additional handling?